python/revng/scripting/cli_project.py

```diff
@@ -0,0 +1,104 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import shutil
+import subprocess
+from tempfile import NamedTemporaryFile, TemporaryDirectory
+from typing import Any, List
+
+import yaml
+
+from revng.model import Binary, dump_diff, load_model_from_string, make_diff
+
+from .project import Project
+
+
+class CLIProject(Project):
+    def __init__(self, revng_executable: str = "", working_dir: str = ""):
+        self.revng_executable = revng_executable
+        self.working_dir = working_dir
+        self.input_binary = ""
+        if not self.working_dir:
+            tmp_dir = TemporaryDirectory()
+            self.working_dir = tmp_dir.name
+        if not self.revng_executable:
+            self.revng_executable = "revng"
+            assert shutil.which(self.revng_executable) is not None
+        super().__init__()
+
+    def import_binary(self, input_binary: str):
+        self.input_binary = input_binary
+        self.analyze("import_binary")
+        with open(f"{self.working_dir}/pipeline-description.yml") as pd:
+            self.pipeline_description = yaml.safe_load(pd)
+        self.set_model(self.model)
+
+    def import_and_analyze(self, input_binary: str) -> None:
+        self.input_binary = input_binary
+        self.analyze("revng-initial-auto-analysis")
+        self.analyze("revng-c-initial-auto-analysis")
+        with open(f"{self.working_dir}/pipeline-description.yml") as pd:
+            self.pipeline_description = yaml.safe_load(pd)
+        self.set_model(self.model)
+
+    def get_artifact(self, artifact_name: str, targets_class: List[Any] = []) -> bytes | str:
+        self._apply_diff()
+
+        args = [
+            self.revng_executable,
+            "artifact",
+            f"--resume={self.working_dir}",
+            artifact_name,
+            self.input_binary,
+        ]
+
+        args.extend(self.get_targets(targets_class, artifact_name))
+
+        return subprocess.run(args, capture_output=True, check=True).stdout
```

Comment:
This only returns `bytes`, you need to check the MIME and return string when needed

Reply:
Is the new [implementation](https://github.com/revng/revng/pull/398/commits/807b125e56e1ddec6ebc53f27424eff862dc839c#diff-4b60e8f32a7d2ba8aa5adb57799f1fa25514f1cd30f4b5bcad0b993ffb7531adR140) ok?

Reply:
Yes